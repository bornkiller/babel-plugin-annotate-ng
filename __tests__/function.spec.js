/**
 * @description - babel-plugin-annotate-ng annotate Function declare unit test
 * @author - bornkiller <hjj491229492@hotmail.com>
 */

const path = require('path');
const babel = require('babel-core');
const annotate = require('../src');
const babelOptions = {
  presets: [],
  plugins: [annotate],
  babelrc: false
};

describe('babel-plugin-annotate-ng Function Declare', function () {
  it('ExportNamedDeclare', function () {
    const fixture = path.resolve(__dirname, '../__fixture__/Function/showcase.controller.js');
    const { code } = babel.transformFileSync(fixture, babelOptions);

    expect(code).toMatchSnapshot();
  });

  it('ExportDefaultDeclaration', function () {
    const fixture = path.resolve(__dirname, '../__fixture__/Function/butterfly.controller.js');
    const { code } = babel.transformFileSync(fixture, babelOptions);

    expect(code).toMatchSnapshot();
  });
});

describe('babel-plugin-annotate-ng Inline Function Declare', function () {
  it('ExportNamedDeclare', function () {
    const fixture = path.resolve(__dirname, '../__fixture__/Function/showcase-inline.controller.js');
    const { code } = babel.transformFileSync(fixture, babelOptions);

    expect(code).toMatchSnapshot();
  });

  it('ExportDefaultDeclaration', function () {
    const fixture = path.resolve(__dirname, '../__fixture__/Function/butterfly-inline.controller.js');
    const { code } = babel.transformFileSync(fixture, babelOptions);

    expect(code).toMatchSnapshot();
  });
});
