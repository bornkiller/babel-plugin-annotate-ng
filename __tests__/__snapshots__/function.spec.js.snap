// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-annotate-ng Function Declare ExportDefaultDeclaration 1`] = `
"ButterflyController.$inject = [\\"$q\\", \\"$sce\\", \\"$animate\\"];

export default function ButterflyController($q, $sce, $animate) {
  this.$q = $q;
  this.$sce = $sce;
  this.$animate = $animate;

  // lifecycle method
  this.ngOnInit = ngOnInit;
  this.ngAfterViewInit = ngAfterViewInit;
  this.ngOnDestroy = ngOnDestroy;

  // function declare
  function ngOnInit() {}

  function ngAfterViewInit() {}

  function ngOnDestroy() {}
}"
`;

exports[`babel-plugin-annotate-ng Function Declare ExportNamedDeclare 1`] = `
"ShowcaseController.$inject = [\\"$q\\", \\"$sce\\", \\"$animate\\"];

export function ShowcaseController($q, $sce, $animate) {
  this.$q = $q;
  this.$sce = $sce;
  this.$animate = $animate;

  // lifecycle method
  this.ngOnInit = ngOnInit;
  this.ngAfterViewInit = ngAfterViewInit;
  this.ngOnDestroy = ngOnDestroy;

  // function declare
  function ngOnInit() {}

  function ngAfterViewInit() {}

  function ngOnDestroy() {}
}"
`;

exports[`babel-plugin-annotate-ng Inline Function Declare ExportDefaultDeclaration 1`] = `
"ButterflyInlineController.$inject = [\\"$q\\", \\"$sce\\", \\"$animate\\"];
export default function ButterflyInlineController($q, $sce, $animate) {
  this.$q = $q;
  this.$sce = $sce;
  this.$animate = $animate;

  // lifecycle method
  this.ngOnInit = ngOnInit;
  this.ngAfterViewInit = ngAfterViewInit;
  this.ngOnDestroy = ngOnDestroy;

  // function declare
  function ngOnInit() {}

  function ngAfterViewInit() {}

  function ngOnDestroy() {}
}"
`;

exports[`babel-plugin-annotate-ng Inline Function Declare ExportNamedDeclare 1`] = `
"ShowcaseInlineController.$inject = [\\"$q\\", \\"$sce\\", \\"$animate\\"];
export function ShowcaseInlineController($q, $sce, $animate) {
  this.$q = $q;
  this.$sce = $sce;
  this.$animate = $animate;

  // lifecycle method
  this.ngOnInit = ngOnInit;
  this.ngAfterViewInit = ngAfterViewInit;
  this.ngOnDestroy = ngOnDestroy;

  // function declare
  function ngOnInit() {}

  function ngAfterViewInit() {}

  function ngOnDestroy() {}
}"
`;
